### ISSUE TITLE
Missing contours in tricontourf with GLMakie

### ISSUE THREAD

**Issue by kimauth** [MakieOrg/Makie.jl#4126]
`tricontourf` misses the top contours when employing GLMakie in some cases. This can be reproduced by reverting the sign of the z data in [the first example from the docs](https://docs.makie.org/stable/reference/plots/tricontourf#examples):
![mwe_tricontourf_glmakie](https://github.com/user-attachments/assets/2e818f02-d1d6-4219-b47c-86ac694d3ce8)

For comparision, the same figure saved with CairoMakie looks correct:
[mwe_tricontourf_cairomakie.pdf](https://github.com/user-attachments/files/16595975/mwe_tricontourf_cairomakie.pdf)

This is generated on a M1 Macbook Pro and `Makie v0.21.5`. It seems to happen when there should be smaller contours with a lower z value on top of larger contours with a higher z value from the cases that I have observed.

Code for reproducing:
```julia
using CairoMakie, GLMakie
using Random
Random.seed!(1234)

x = randn(50)
y = randn(50)
z = -sqrt.(x .^ 2 .+ y .^ 2) .+ 0.1 .* randn.()

# https://docs.makie.org/stable/reference/plots/tricontourf
f = Figure(; size=(1200, 450))
ax = Axis(f[1,1]; title="original example")
tr = tricontourf!(ax, x, y, z)
scatter!(x, y, color = z, strokewidth = 1, strokecolor = :black)
Colorbar(f[1, 2], tr)
# add 2nd plot with inverted z values
ax_neg = Axis(f[1,3]; title="reverse sign of z data")
tr_neg = tricontourf!(x, y, -z)
scatter!(x, y, color = -z, strokewidth = 1, strokecolor = :black)
Colorbar(f[1, 4], tr_neg)

GLMakie.activate!()
save("mwe_tricontourf_glmakie.png", f)

CairoMakie.activate!()
save("mwe_tricontourf_cairomakie.pdf", f)
```

**Comment by SimonDanisch** [comment]
@jkrumbiegel, @DanielVandH looks like the polygons aren't hollow?
I think they should be, or we need to sort/depth_shift it by enclosing area or something like that?

**Comment by jkrumbiegel** [comment]
Yeah they should be hollow. Contourf does that, not sure about the implementation here.
